crates/sui-indexer-alt/src/handlers/coin_balance_buckets.rs

```diff
@@ -159,9 +160,14 @@ impl Processor for CoinBalanceBuckets {
 
 #[async_trait::async_trait]
 impl Handler for CoinBalanceBuckets {
+    type Store = Db;
+
     const PRUNING_REQUIRES_PROCESSED_VALUES: bool = true;
 
-    async fn commit(values: &[Self::Value], conn: &mut db::Connection<'_>) -> Result<usize> {
+    async fn commit<'a>(
+        values: &[Self::Value],
+        conn: &mut <Self::Store as Store>::Connection<'a>,
```

Comment:
Within handlers, I think it's fine (preferred) to refer to the `Connection` directly on the concrete store type:

```suggestion
        conn: &mut Db::Connection<'a>,
```

or (if there are two associated `Connection` types for `Db`):

```suggestion
        conn: &mut db::Connection<'a>,
```

But probably if that's the case the better thing to do would be to remove the previously introduced associated type, because it was probably trying to achieve the same thing that the `Store` trait is achieving now.

Reply:
The other connection type comes from `pub type Connection<'p> = PooledConnection<'p, AsyncPgConnection>;` in `sui_pg_db`, so a convenience type alias? Even if we were to rename this though, Rust still complains about ambiguous associated type if we try to reference `Db::Connection` in the handler. Actually, I don't think sui_pg_db::Connection is associated at all to the Db, the association comes from implementing Store on Db? So rust is complaining that there's a possibility for multiple Store to be implemented for diff Connection. Which is why when we reference `db::Connection` the complaint goes away. This works in our favor since we want handler impl to refer to the concrete type anyways